data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsDataFactory.java

```diff
@@ -33,38 +35,87 @@ public MetricsDataFactory(MetricsSystem metricsSystem) {
 
-  public BaseMetricData getMetricData(MetricsDataClient metricsClient) {
+  public List<BaseMetricData> getMetricData() {
+    List<BaseMetricData> metricList = new ArrayList<>();
     if (metricsSystem instanceof PrometheusMetricsSystem) {
-      if (metricsClient == MetricsDataClient.VALIDATOR) {
-        return extractValidatorData((PrometheusMetricsSystem) metricsSystem);
-      }
+      Map<String, Object> values = getStringObjectMap((PrometheusMetricsSystem) metricsSystem);
+      metricList.add(extractBeaconNodeData(values));
+      metricList.add(extractValidatorData(values));
     } else {
       LOG.error("Prometheus metric system not found.");
+      metricList.add(
+          new MinimalMetricData(
+              PROTOCOL_VERSION, System.currentTimeMillis(), "minimal", metricsSystem));
     }
-    return new MinimalMetricData(
-        PROTOCOL_VERSION, System.currentTimeMillis(), "minimal", metricsSystem);
+    return metricList;
   }
 
-  private static BaseMetricData extractValidatorData(
-      PrometheusMetricsSystem prometheusMetricsSystem) {
+  private static BaseMetricData extractBeaconNodeData(Map<String, Object> values) {
+    final long disk_beaconchain_bytes_total;
+    final long network_libp2p_bytes_total_receive;
+    final long network_libp2p_bytes_total_transmit;
+    final int network_peers_connected;
+    final long sync_beacon_head_slot;
+    final String clientVersion;
+
+    if (values.containsKey("head_slot")) {
+      sync_beacon_head_slot =
+          ((Double) ((Observation) values.get("head_slot")).getValue()).longValue();
+    } else {
+      sync_beacon_head_slot = 0;
+    }
+    if (values.containsKey("teku_version")) {
+      clientVersion = ((Observation) values.get("teku_version")).getValue().toString();
+    } else {
+      clientVersion = "";
+    }
+    if (values.containsKey("filesystem_size_bytes")) {
+      disk_beaconchain_bytes_total =
+          ((Double) ((Observation) values.get("filesystem_size_bytes")).getValue()).longValue();
+    } else {
+      disk_beaconchain_bytes_total = 0;
+    }
+    if (values.containsKey("network_receive_bytes_total")) {
+      network_libp2p_bytes_total_receive =
+          ((Double) ((Observation) values.get("network_receive_bytes_total")).getValue())
+              .longValue();
+    } else {
+      network_libp2p_bytes_total_receive = 0;
+    }
+    if (values.containsKey("network_transmit_bytes_total")) {
+      network_libp2p_bytes_total_transmit =
+          ((Double) ((Observation) values.get("network_transmit_bytes_total")).getValue())
+              .longValue();
+    } else {
+      network_libp2p_bytes_total_transmit = 0;
+    }
+    if (values.containsKey("peer_count")) {
+      network_peers_connected =
+          ((Double) ((Observation) values.get("peer_count")).getValue()).intValue();
+    } else {
+      network_peers_connected = 0;
+    }
+    return new BeaconNodeMetricData(
+        PROTOCOL_VERSION,
+        System.currentTimeMillis(),
```

Comment:
Would be good to inject a `TimeProvider` instance and use that to get the current time rather than calling `System.currentTimeMillis()`.  You can then pass in a `StubTimeProvider` in tests and actually check it's using the right time.  Mostly just makes it consistent with other parts of the system though.